core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -118,14 +121,89 @@ oppia.directive('questionPlayer', [
       controllerAs: '$ctrl',
       controller: [
-        '$rootScope', '$scope', 'QuestionPlayerBackendApiService',
+        'HASH_PARAM', '$scope', '$sce', '$rootScope', '$location',
+        '$sanitize', '$window', 'HtmlEscaperService',
+        'QuestionPlayerBackendApiService',
         function(
-            $rootScope, $scope, QuestionPlayerBackendApiService) {
+            HASH_PARAM, $scope, $sce, $rootScope, $location,
+            $sanitize, $window, HtmlEscaperService,
+            QuestionPlayerBackendApiService) {
           var ctrl = this;
-          $scope.questionPlayerConfig = ctrl.getQuestionPlayerConfig();
+          var questionPlayerConfig = ctrl.getQuestionPlayerConfig();
+          $scope.resultsLoaded = false;
           ctrl.currentQuestion = 0;
           ctrl.totalQuestions = 0;
           ctrl.currentProgress = 0;
+          ctrl.totalScore = 0.0;
+
+          var VIEW_HINT_PENALTY = 0.1;
+          var WRONG_ANSWER_PENALTY = 0.1;
+
+          var getStaticImageUrl = function(url) {
+            return UrlInterpolationService.getStaticImageUrl(url);
+          };
+
+          ctrl.getActionButtonOuterClass = function(actionButtonType) {
+            var className = getClassNameForType(actionButtonType);
+            if (className) {
+              return className + 'outer';
+            }
+            return '';
+          };
+
+          ctrl.getActionButtonInnerClass = function(actionButtonType) {
+            var className = getClassNameForType(actionButtonType);
+            if (className) {
+              return className + 'inner';
+            }
+            return '';
+          };
+
+          ctrl.getActionButtonIconHtml = function(actionButtonType) {
+            var iconHtml = '';
+            if (actionButtonType === 'BOOST_SCORE') {
+              iconHtml = '<img class="action-button-icon" src="' +
+              getStaticImageUrl('/icons/rocket@2x.png') + '"/>';
+            } else if (actionButtonType === 'RETRY_SESSION') {
+              iconHtml = '<i class="material-icons md-36 ' +
+              'action-button-icon">&#xE5D5</i>';
+            } else if (actionButtonType === 'DASHBOARD') {
+              iconHtml = '<i class="material-icons md-36 ' +
+              'action-button-icon">&#xE88A</i>';
+            }
+            return $sce.trustAsHtml($sanitize(iconHtml));
+          };
+
+          ctrl.performAction = function(actionButton) {
+            if (actionButton.type === 'BOOST_SCORE') {
+              boostScoreModal();
+            } else if (actionButton.url) {
+              $window.location.href = actionButton.url;
+            }
+          };
+
+          ctrl.showActionButtonsFooter = function() {
+            return (questionPlayerConfig.resultActionButtons &&
+              questionPlayerConfig.resultActionButtons.length > 0);
+          };
+
+          var boostScoreModal = function() {
+            // Will open a boost score modal that explains the worst skill
```

Comment:
Maybe add a TODO here? Or file an issue, so that it is tracked.

Reply:
The issue is part of the overall results page issue - https://github.com/oppia/oppia/issues/6347

Reply:
Ah thanks for the reference.